File: dotnet/AzureApps/AzureApps/Modules/Comics/Drivers/Explosm.cs
using System.Text;
using dk.schau.AzureApps.Commands;
using dk.schau.AzureApps.Modules.Comics.DomainModel;

namespace dk.schau.AzureApps.Modules.Comics.Drivers
{
	public class Explosm : IComicsDriver
	{
		readonly Fetcher _fetcher;

		public Explosm(Fetcher fetcher)
		{
			_fetcher = fetcher;
		}

		public string Execute(string url, string name)
		{
			var response = _fetcher.ExecuteAsync(url).Result;
			var html = Encoding.UTF8.GetString(response);
			var pos = html.IndexOf("div id=\"comic-wrap\"");
			if (pos >= 0)
			{
				pos = html.IndexOf("flex-video", pos);
				if (pos >= 0)
				{
					return $"<p>Today is a video. Click <a href=\"{url}\">here</a> to see it :-)</p>";
				}
			}

			pos = html.IndexOf("img id=\"main-comic\" src");
			var startPos = pos + 25;
			var endPos = html.IndexOf('"', startPos);
			var link = html.Substring(startPos, endPos - startPos);
			if (link.StartsWith("//"))
			{
				link = $"http:{link}";
			}

			return $"<p><img src=\"{link}\" alt=\"{name}\" width=\"900\" /></p><p><br /></p>";
		}
	}
}

File: VpsApps/go/programs/internal/comics/explosm.go
package comics

import (
	"internal/common"
	"strings"
)

// Explosm Handle Explosm comic
func Explosm() {
	url := "http://explosm.net"
	data := FetchURL(url)
	if data == nil {
		return
	}

	common.HTMLHeader("Explosm", url)
	text := string(data)
	pos := strings.Index(text, "div id=\"comic-wrap\"")
	if pos >= 0 {
		pos = IndexAt(text, "flex-video", pos)
		if pos >= 0 {
			body := "<p>Today is a video. Click <a href=\"" + url + "\">here</a> to see it :-)</p>"
			common.HTMLLineBreak(body)
			return
		}
	}

	pos = strings.Index(text, "img id=\"main-comic\" src")
	startPos := pos + 25
	endPos := IndexAt(text, "\"", startPos)
	link := text[startPos:endPos]
	if link[:2] == "//" {
		link = "http:" + link
	}

	common.HTMLAddComic("Explosm", link, "900")
	common.HTMLLineBreak("")
}
